hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/kafka/KafkaFacade.java

```diff
@@ -173,35 +164,24 @@ public ProjectTopics findTopicByNameAndProject(Project project, String topicName
    *
    */
   public List<SharedTopics> findSharedTopicsByProject(Integer projectId) {
-    TypedQuery<SharedTopics> query = em.createNamedQuery(
-        "SharedTopics.findByProjectId",
-        SharedTopics.class);
-    query.setParameter("projectId", projectId);
-    return query.getResultList();
+    return em.createNamedQuery("SharedTopics.findByProjectId", SharedTopics.class)
```

Comment:
@kai-chi instead of a list, we might need to return a CollectionInfo, as for pagination we need to return the count as well https://github.com/logicalclocks/hopsworks/blob/master/hopsworks-common/src/main/java/io/hops/hopsworks/common/dao/jobs/description/JobFacade.java#L208 
That means KafkaFacade would extend the AbstractFacade. 

How does the solution presented here handle pagination?

Reply:
@tkakantousis, pagination is handled in `CollectionsBuilder.java` which does it on a java stream. 
I was thinking about extending `AbstractFacade` but didn't know which class would be the type parameter since `KafkaFacade` deals with multiple tables?

Reply:
We can omit the type parameter, it's not enforced.

Reply:
solved by splitting to multiple facades